data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp.api.configuration;
+
+//import org.eclipse.edc.jsonld.transformer.JsonLdTransformerRegistry;
+import org.eclipse.edc.protocol.dsp.api.configuration.auth.ClaimTokenRequestFilter;
+import org.eclipse.edc.protocol.dsp.api.configuration.serdes.ObjectMapperProvider;
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.runtime.metamodel.annotation.Provides;
+//import org.eclipse.edc.runtime.metamodel.annotation.Requires;
+import org.eclipse.edc.spi.iam.IdentityService;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.web.spi.WebServer;
+import org.eclipse.edc.web.spi.WebService;
+import org.eclipse.edc.web.spi.configuration.WebServiceConfigurer;
+import org.eclipse.edc.web.spi.configuration.WebServiceSettings;
+
+import static org.eclipse.edc.jsonld.JsonLdExtension.TYPE_MANAGER_CONTEXT_JSON_LD;
+
+/**
+ * Provides the configuration for the Dataspace Protocol API context. Creates the API context
+ * using {@link #DEFAULT_PROTOCOL_PORT} and {@link #DEFAULT_PROTOCOL_API_PATH}, if no respective
+ * settings are provided. Configures the API context to allow Jakarta JSON-API types as endpoint
+ * parameters and registers a filter that validates the authorization header using the
+ * IdentityService.
+ */
+@Extension(value = DspApiConfigurationExtension.NAME)
+@Provides({ DspApiConfiguration.class })
+//@Requires({ JsonLdTransformerRegistry.class })
+public class DspApiConfigurationExtension implements ServiceExtension {
+    
+    public static final String NAME = "Dataspace Protocol API Configuration Extension";
+    
+    public static final String CONTEXT_ALIAS = "protocol";
+    
+    public static final String DEFAULT_DSP_WEBHOOK_ADDRESS = "http://localhost:8282/api/v1/dsp";
+    public static final String DSP_WEBHOOK_ADDRESS = "dsp.webhook.address";
```

Comment:
`edc.` prefix. 
one is enough, or should we have one for every context (catalog, negotiation, transfer)?

Reply:
I would leave it at one for the time being. We can still switch to using multiple later on if necessary.